### ISSUE TITLE
"unknown proxy name: 'Cursor'" when using rust

### ISSUE THREAD

**Issue by LeoCatsune** [cursor/cursor#549]
For some reason, `ARGV0` gets overwritten when running commands, which affects several commands in the rust toolchain such as `cargo` and `rustup`.

The same issue can be observed in similar situations at GitSquared/edex-ui#1124 and onivim/oni2#3830.

**Comment by truell20** [comment]
Thanks for flagging this. Where does this happen in the UI? Or are you talking about the integrated terminal.

**Comment by ProtogenDelta** [comment]
This is in the integrated terminal, from memory.

**Comment by jzanecook** [comment]
I was able to override this for my own session by running `unset ARGV0`

Wherever it is setting the `ARGV0` is where the issue starts.

What is the purpose of this `ARGV0` variable being set to the cursor application? Is it for the cursor command to pick up?

**Comment by cs01** [comment]
+1 just hit this too.  `unset ARGV` fixed it, but would be great not to have to do that.
```
Version: 0.25.1
VSCode Version: 1.85.1
Commit: 424d9aa49a015f43a2299e679576231a75a34fd0
Date: 2024-02-03T21:07:19.472Z
Electron: 25.9.7
ElectronBuildId: undefined
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 6.5.0-15-generic
```

**Comment by definitelynobody** [comment]
This also happens when tasks are launched that run cargo and rust directly. Unsetting this from ~/.zshrc, etc doesn't work for me.

**Comment by bwks** [comment]
still happening in v0.40.0

**Comment by tumluliu** [comment]
still happening in v0.42.2

**Comment by zhoujingya** [comment]
still happening in v0.42.2

**Comment by eliheuer** [comment]
I had the same issue in v0.42.3, `unset ARGV0` fixed it.

I only get this on Ubuntu 24 Linux, when using macOS it is not a problem.

**Comment by raghav-rama** [comment]
This affects virtual environments in python too. I get module not found error whereas it is clearly installed in the `venv`
![image](https://github.com/user-attachments/assets/1841185e-244a-4d61-aa2f-5292236d8b81)

**Comment by thebarkingdog-yh** [comment]
v0.43.5 still has this issue, and the Python virtual environment is not activated.

**Comment by jzanecook** [comment]
This is very strange... While I only seem to have the issue on Cursor AFAIK, it looks like the cause might not be from cursor itself.

They've added a method of viewing what extensions want to modify the terminal environment (since when, not sure). Below is a screenshot:

![image](https://github.com/user-attachments/assets/83540144-a38c-4e1a-b52a-cf388872bf0e)

Clicking on the "Show Environment Contributions" gives me a file like so:

```
# Terminal Environment Changes

# [...]  Removed some non-topic text here

## Extension: ms-python.python

- `ARGV0=/opt/appimages/cursor.AppImage`
- `PATH=/home/z/.pyenv/versions/3.11.4/bin:${env:PATH}`
```

What I notice from this... is that ARGV0 appears to be coming from the ms-python.python package.

So what I'm saying is... perhaps it was Microsoft all along!

But why is ARGV0 empty by default in vs code?
Why isn't the python package setting it there (verified my VS Code has ms-python.python as well)?

Is it sabotage?

Just figured since I get emails from this issue occasionally I would share. Good luck, y'all. :D

**Comment by nazdridoy** [comment]
add this to your bashrc/zshrc

```bash
if [[ "$ARGV0" == "/opt/cursor-bin/cursor-bin.AppImage" ]]; then
  unset ARGV0
fi
```

**Comment by bhouser** [comment]
This is not just a rust issue. This problem prevents me from running tests in my typescript repo using `pnpm`.

**Comment by wiltaylor** [comment]
This is really strange. I get the above error even after unsetting ARG0. However if I run other commands like just it doesn't give me an error. I just put all of my build stuff in a just script but this bug really should be resolved by now.

**Comment by bengabp** [comment]
> add this to your bashrc/zshrc
> 
> if [[ "$ARGV0" == "/opt/cursor-bin/cursor-bin.AppImage" ]]; then
>   unset ARGV0
> fi

This worked for me

**Comment by GrumpyMetalGuy** [comment]
I ended up adding

```json
    "terminal.integrated.env.linux": {
        "ARGV0": null,
    },
```

to my settings.json, and it seems to have helped.

**Comment by r3b311i0n** [comment]
> I ended up adding
> 
>     "terminal.integrated.env.linux": {
>         "ARGV0": null,
>     },
> to my settings.json, and it seems to have helped.

This is what finally worked for me. None of the other solutions helped.

**Comment by Jongy** [comment]
I just asked cursor to prefix all commands it runs with `unset ARGV0`, also seems to work.

**Comment by Yukigaru** [comment]
Still an issue in the latest:
```
Version: 0.49.4
VSCode Version: 1.96.2
OS: Linux x64 6.8.0-52-generic
```
Workarounds work.

**Comment by evgenyneu** [comment]
The `unset ARGV0` in .zshrc did not work for me. When I open the new integrated terminal, the ARGV0 variable still has path to cursor image cursor.AppImage.

**Comment by gbaian10** [comment]
I have used `poetry` before, and there was an old virtual environment folder for the project in `.cache/pypoetry`.
Even though I didn’t select the .venv in .cache in `Cursor`, it still caused the above solution to fail (this issue doesn’t occur in other new projects, where the above solution works).
Once I deleted that Poetry virtual environment folder, the above command started working.
If you're someone migrating from `poetry` to `uv`, you might want to try this solution.

**Comment by zhouqiji** [comment]
Same bug, when I compile files with gcc/cc.

**Comment by danieleades** [comment]
This issue is a massive blocker for adoption. Not so much the issue itself, but the fact something so important, and so minor to fix, has been left unaddressed for so long. Not confidence inspiring.

**Comment by brickfrog** [comment]
yeah.... I keep running into things like this and finding years old open issues is a bit discouraging

**Comment by Nathanjp91** [comment]
Just run into this. Still present

**Comment by shubhamprasad0** [comment]
Still seeing this while using Rust

`Version: 1.4.5
VSCode Version: 1.99.3
Commit: af58d92614edb1f72bdd756615d131bf8dfa5290
Date: 2025-08-13T02:08:56.371Z
Electron: 34.5.8
Chromium: 132.0.6834.210
Node.js: 20.19.1
V8: 13.2.152.41-electron.0
OS: Linux x64 6.8.0-64-generic`
